aclk/agent_cloud_link.c

```diff
@@ -1117,11 +1121,17 @@ void aclk_shutdown()
 
 int aclk_init(ACLK_INIT_ACTION action)
 {
-    UNUSED(action);
-
+//    UNUSED(action);
     static int init = 0;
     int rc;
 
+    if (action == ACLK_REINIT) {
+        unsigned long delay = aclk_reconnect_delay(1);
+        info("Retrying to establish the ACLK connection in %.3f seconds", delay / 1000.0);
+        sleep_usec(USEC_PER_MS * delay);
```

Comment:
A reconnection is different from an initial connection. If we sleep for some number of seconds here it is not clear that the thread should be blocked and the event loop for libwebsockets suspended.